x-pack/plugins/security_solution/server/lib/detection_engine/rule_schema/model/rule_alert_type.ts

```diff
@@ -6,14 +6,13 @@
  */
 
 import { ruleTypeMappings } from '@kbn/securitysolution-rules';
-import type { SanitizedRule } from '@kbn/alerting-plugin/common';
-import type { PartialRule } from '@kbn/alerting-plugin/server';
+import type { ResolvedRule, Rule } from '@kbn/alerting-plugin/server';
 import type { RuleParams } from './rule_schemas';
 
-export type RuleAlertType = SanitizedRule<RuleParams>;
+export type RuleAlertType = Rule<RuleParams>;
 
 export const isAlertType = (
-  partialAlert: PartialRule<RuleParams>
+  partialAlert: Partial<ResolvedRule<RuleParams>>
```

Comment:
What is the reason and need for changing the rule types here:

- `SanitizedRule` to `Rule`
- `PartialRule` to `ResolvedRule`

How versioning is related to that?

These are not just renamings, aren't they?

Reply:
Reverting this for now but be advised, these types will need to change as part of the HTTP versioning effort and that needs to be Security Solution's responsibility. RAM does not have the bandwidth to continue updating your team's code.

We will be pushing ts-ignores in the future if we have to.